luaa.h

```diff
@@ -153,6 +153,33 @@ luaA_getopt_number(lua_State *L, int idx, const char *name, lua_Number def)
     return def;
 }
 
+static inline double
+luaA_checknumber_range(lua_State *L, int n, double min, double max)
+{
+    double result = lua_tonumber(L, n);
+    if (result < min || result > max)
+        luaA_rangerror(L, n, min, max);
+    return result;
+}
+
+static inline lua_Number
+luaA_optnumber_range(lua_State *L, int narg, lua_Number def, double min, double max)
+{
+    if (lua_isnoneornil(L, narg))
+        return def;
+    return luaA_checknumber_range(L, narg, min, max);
+}
+
+static inline int
+luaA_getopt_number_range(lua_State *L, int idx, const char *name, lua_Number def, double min, double max)
+{
+    lua_getfield(L, idx, name);
+    if (lua_isnil(L, -1) || lua_isnumber(L, -1))
+        def = luaA_optnumber_range(L, -1, def, min, max);
+    lua_pop(L, 1);
+    return def;
```

Comment:
Could you turn all of the `double`s here into `lua_Number`? I don't really have a good reason for this, but the rest of the file uses `lua_Number` as well.